tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -875,6 +875,13 @@ DeclResultIdMapper::getDeclSpirvInfo(const ValueDecl *decl) const {
 SpirvInstruction *DeclResultIdMapper::getDeclEvalInfo(const ValueDecl *decl,
                                                       SourceLocation loc,
                                                       SourceRange range) {
+  if (hlsl::IsHLSLDynamicResourceType(decl->getType())) {
```

Comment:
Can you explain to me why this will impact `ResourceDescriptorHeap` and not other types?

Reply:
To be honest though I did also rely on just seeing that that was all code that was added in the big SM 6.6 patch, and that all tests passed still passed when I added this, but if that feels like logic that's a bit too shaky I can change it to string checking for `"ResourceDescriptorHeap"`.

Reply:
In my opinion it's better to use the existing `IsHLSLDynamicResourceType`, which seems to be explicitly for checking for this type, than using a string comparison.